source/FreeRTOS_IPv4.c

```diff
@@ -260,6 +312,13 @@ enum eFrameProcessingResult prvAllowIPPacketIPv4( const struct xIP_PACKET * cons
             /* Can not handle, unknown or invalid header version. */
             eReturn = eReleaseBuffer;
         }
+        else if( ( pxIPPacket->xEthernetHeader.usFrameType == ipIPv4_FRAME_TYPE ) &&
```

Comment:
`pxIPPacket->xEthernetHeader.usFrameType == ipIPv4_FRAME_TYPE`

This checks seems to be redundant. `prvAllowIPPacketIPv4` is only called when `pxIPPacket->xEthernetHeader.usFrameType` is `ipIPv4_FRAME_TYPE` from the `prvProcessIPPacket()`.

Reply:
Yes good idea, it was tested already. I also adapted the test in `FreeRTOS_IPv4_utest.c`.